src/player.cpp

```diff
@@ -9610,32 +9610,41 @@ bool player::wield( item& target )
 
     int mv = 0;
 
+    bool target_removed = false;
+    item target_copy = target;
     if( is_armed() ) {
+        inventory old_inv = inv;
```

Comment:
It is more overhead than necessary (whoops, an oversight), but how does it look like a hack? If I removed that line and replaced `inv = old_inv` with
`if( target_removed ) {`
`    i_add( target_copy )`
`}`, would it still look hack-ish?